test/test_sparse.py

```diff
@@ -4233,6 +4233,14 @@ def explicit_to_sparse(x):
                 r2 = explicit_to_sparse(t)
                 self.assertEqual(r2, r)
 
+        # extra tests
+        if (from_layout, to_layout) == (torch.sparse_csr, torch.sparse_bsr):
+            # See gh-90910
+            t = torch.tensor([[0, 0, 1, 0], [0, 1, 0, 0]], dtype=dtype, device=device).to_sparse_csr()
+            r = t.to_sparse_bsr(2, 2)
+            torch._validate_sparse_compressed_tensor_args(r.crow_indices(), r.col_indices(), r.values(), r.shape, r.layout)
+            self.assertEqual(r, t)
```

Comment:
Let's probably add interesting cases with different nnz per blocks?

Reply:
The diagnosis of the fixed issue is in https://github.com/pytorch/pytorch/issues/90910#issuecomment-1352896532 and the given special case is a sufficient representation of the bug. Also, the addmv tests use inputs that touch this bug. The bug was not due to a particular nnz size but due to a certain position of the first non-zero element in the block, so variations of nnz here would have only a secondary effect on testing this bug.

Reply:
Note that the bug was silent because we don't have tests for unsafe constructors and "uncoalesced CSR" works with many existing algorithms in PyTorch. We already have lots of tests with variable nnz that likely touch this bug but these were not discovered nevertheless. In follow-up PRs, we should ensure that all unsafe constructions of sparse tensors are tested via the proper validate function.